Sources/Engine/WSEngine.swift

```diff
@@ -125,9 +125,7 @@ FrameCollectorDelegate, HTTPHandlerDelegate {
             let wsReq = HTTPWSHeader.createUpgrade(request: request, supportsCompression: framer.supportsCompression(), secKeyValue: secKeyValue)
             let data = httpHandler.convert(request: wsReq)
             transport.write(data: data, completion: {_ in })
-        case .waiting:
-            break
-        case .failed(let error):
+        case .waiting(let error), .failed(let error):
```

Comment:
I don't think that engine should depend on transport logic. If we want to say that some waiting states is actually errors then let's just report failed state from transport.

Reply:
Good, the error is now propagated with a failed state. I think we should keep the new version of the `.waiting` state to ensure we don't overlook errors from the new transport. The NWConnection's `.waiting` state also has associated errors with it.

Reply:
I don't think so. It doesn't make sense to say that waiting state is might actually be failed state - it looks like improper implementation of the transport

Reply:
So you want to reroute `NWErrors` from `NWConnection`'s .waiting state to transport's .failed state, right?

Reply:
I suggest to rollback changes for WSEngine. So it will be untouched